### PR TITLE
Modular daemon: fix run daemon foreground  issues

### PR DESCRIPTION
**Pull request by yafu-1**
Signed-off-by: Yan Fu <yafu@redhat.com>

# Format of PR title < sub-system: summary >
e.g
*virsh_migrate: Fix unsupported direct socket mode issue*

# Check lists by category
## If the PR is new cases
- [ ] Description of the cases
- [ ] Links of libvirt features, libvirt bugs or case IDs
- [ ] Test results

## If the PR is bug cases
- [ ] Bug descriptions or bug links
- [ ] Test results

## If the PR is a trivial fix
It it is the fix of typos, comments or documents, the description of PR could be skipped.

### CODE REVIEW

**Comment by yafu-1** [comment]
Test results:
$avocado run daemon.functional --vt-type libvirt
WARNING:root:No python imaging library installed. Screendump and Windows guest BSOD detection are disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
WARNING:root:No python imaging library installed. PPM image conversion to JPEG disabled. In order to enable it, please install python-imaging or the equivalent for your distro.
JOB ID     : f67c67df125c71481154af80b5c19243688889d3
JOB LOG    : /root/avocado/job-results/job-2021-07-02T02.49-f67c67d/job.log
 (01/10) type_specific.io-github-autotest-libvirt.daemon.functional.no_opt: PASS (75.35 s)
 (02/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_help: PASS (6.76 s)
 (03/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_version: PASS (5.28 s)
 (04/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_timeout_3: PASS (68.72 s)
 (05/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_timeout_3_live_vm: PASS (92.38 s)
 (06/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_config: PASS (77.06 s)
 (07/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_pid: PASS (75.02 s)
 (08/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_invalid_option: PASS (6.18 s)
 (09/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_invalid_timeout_short: PASS (5.29 s)
 (10/10) type_specific.io-github-autotest-libvirt.daemon.functional.opt_invalid_timeout_long: PASS (7.28 s)
RESULTS    : PASS 10 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

File: libvirt/tests/src/daemon/daemon_functional.py [hunk 1]
```diff
@@ -41,7 +43,10 @@ def check_version(params):
         Check whether the output is libvirtd version.
         """
         expected_version = params.get('expected_version', 'no') == 'yes'
-        is_version = log[0].startswith('libvirtd (libvirt)')
+        if utils_split_daemons.is_modular_daemon():
+            is_version = log[0].startswith('virtqemud (libvirt)')
+        else:
+            is_version = log[0].startswith('libvirtd (libvirt)')
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/daemon_functional.py:49]:
```suggestion
        is_version = log[0].startswith('{} (libvirt)'.format(utils_libvirtd.Libvirtd().service_list[0]))
```

File: libvirt/tests/src/daemon/daemon_functional.py [hunk 2]
```diff
@@ -75,8 +80,10 @@ def check_pid_file():
 
         with open(pid_path) as pid_file:
             pid = int(pid_file.readline())
-
-        result = process.run('pgrep libvirtd', ignore_status=True, shell=True)
+        if utils_split_daemons.is_modular_daemon():
+            result = process.run('pgrep virtqemud', ignore_status=True, shell=True)
+        else:
+            result = process.run('pgrep libvirtd', ignore_status=True, shell=True)
```
Comment (Yingshun) [inline libvirt/tests/src/daemon/daemon_functional.py:86]: same as above.

File: libvirt/tests/src/daemon/daemon_functional.py (hunk 1 above)
Comment (yafu-1) [reply libvirt/tests/src/daemon/daemon_functional.py]: Updated.

File: libvirt/tests/src/daemon/daemon_functional.py (hunk 2 above)
Comment (yafu-1) [reply libvirt/tests/src/daemon/daemon_functional.py]: Updated.

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by chloerh — APPROVED** [review]

**Pull request merged**